### PR TITLE
1645 jr est val

### PR DESCRIPTION
**Pull request by RoyPriceCS50**
## Description
Trello ticket [1645](https://trello.com/c/yJnhDeEj)

Added validation for estimates by job role dataset.

Also in this PR, I spotted the reports path when validation functions are split into multiple calls wasn't right in the imputation validation so fixed that.

I moved the boundaries for the care worker national percentage check. See Trello so working. This should cover ticket [1633](https://trello.com/c/fQM5pI3q).

I deleted the check that ascwds worker counts must be less than estimate filled posts. We merge ascwds and pir, so if pir is the replacement then it is not relative to worker records anymore so can be very different. We don't mark these rows so they can't be factored into the check.

## Testing
- [x] Unit tests passing
- [x] Successful [Step Function run](https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/v2/executions/details/arn:aws:states:eu-west-2:856699698263:execution:1645-jr-est-val-Ind-CQC-Filled-Post-Estimates-By-Role:3154cb5a-9951-4bd9-9357-35845d28c8bb)

No outputs since it's validation.

## Checklist (delete if not relevant)
- [x] Unit tests added/amended
- [x] Docstrings added/updated
- [ ] Updated CHANGELOG
- [x] Code reviewed by AI
- [x] Moved Trello ticket to PR column

## Reviewer checklist
- [ ] PR solves the Trello ticket requirements
- [ ] Outputs appear reasonable and understandable
- [ ] The overall approach is appropriate and not over-engineered
- [ ] No obvious scalability, performance or interdependency concerns
- [ ] Tests appropriately cover the main behaviour/edge cases
- [ ] Naming and structure are broadly understandable and consistent
- [ ] Docstrings are sufficient for future maintenance
- [ ] Docstrings cover non-obvious behaviour

### CODE REVIEW

File: projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/fargate/validate_04_estimates.py [hunk 1]
```diff
@@ -1,36 +1,105 @@
...
+        ),
+        PartitionKeys.year: "Int64",
+    }
+)
+
+CQC_EARLIEST_IMPORT_DATE = date(2013, 3, 1)
+
 req_pcts = {
-    MainJobRoleLabels.care_worker: (0.59, 0.69),
+    MainJobRoleLabels.care_worker: (0.52, 0.69),
```
Comment (AchyutRatkanthiwar) [inline projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/fargate/validate_04_estimates.py:101]: Do we need to change higher limit as well?

File: projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/fargate/validate_04_estimates.py [hunk 2]
```diff
@@ -166,6 +505,28 @@ def estimates_percentage_expressions(
     )
 
 
+def make_convert_col_to_integers_preprocessor(
```
Comment (AchyutRatkanthiwar) [inline projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/fargate/validate_04_estimates.py]: This can be moved to actions.py and its test to test_validation_actions.py

File: projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/tests/fargate/test_validate_04_estimates.py [hunk 3]
```diff
@@ -19,22 +22,44 @@
...
             IndCqcColumns.estimate_filled_posts_by_job_role_manager_adjusted: pl.Float32,
             IndCqcColumns.estimate_filled_posts_from_all_job_roles: pl.Float32,
+            IndCqcColumns.difference_estimate_filled_posts_and_from_all_job_roles: pl.Float32,
             IndCqcColumns.main_job_group_labelled: pl.String,
+            PartitionKeys.year: pl.String,
         }
         source_rows = [
-            ("1", "1-001", date(2026, 1, 1), 100.0, "source", MainJobRoleLabels.care_worker,    40.0, 100.0, JobGroupLabels.direct_care),
-            ("2", "1-002", date(2026, 1, 1), 100.0, "source", MainJobRoleLabels.support_worker, 30.0, 100.0, JobGroupLabels.direct_care),
+            ("1", "1-001", date(2026, 1, 1), 100.0, "non-residential", 1, MainJobRoleLabels.care_worker,    40, "Rule", 0.1, 0.1, 10.0, "size", 0.1, 0.1, "source", 10.0, 10.0, 10.0, 10.0, 10.0, JobGroupLabels.direct_care, "2026"),
```
Comment (AchyutRatkanthiwar) [inline projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/tests/fargate/test_validate_04_estimates.py]: use PrimaryServiceType.non_residential in test data

**Review by AchyutRatkanthiwar — APPROVED** [review]
Just few small comments. looks good 😄

File: projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/fargate/validate_04_estimates.py (hunk 1 above)
Comment (RoyPriceCS50) [reply projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/fargate/validate_04_estimates.py:101]: This is just to make validation succeed. Ticket 1633 will check if this is appropriate long term.
